docs/site/Loopback-component-authorization.md

```diff
@@ -0,0 +1,346 @@
+  return MethodDecoratorFactory.createDecorator<AuthorizationMetadata>(
+    AUTHORIZATION_METADATA_ACCESSOR,
+    {
+      permissions: permissions || [],
+    },
+  );
+}
+```
+
+For error handling keys, lets create an enum.
+
+{% include code-caption.html content="/src/authorization/error-keys.ts" %}
+
+```ts
+export const enum AuthorizeErrorKeys {
+  NotAllowedAccess = 'Not Allowed Access',
+}
+```
+
+Finally, we put everything together in one index file.
+
+{% include code-caption.html content="/src/authorization/index.ts" %}
+
+```ts
+export * from './component';
+export * from './types';
+export * from './keys';
+export * from './error-keys';
+export * from './decorators/authorize.decorator';
+export * from './providers/authorization-metadata.provider';
+export * from './providers/authorization-action.provider';
+```
+
+That is all for the authorization component. You can create all of the above
+into a loopback extension as well. Everything remains the same. Refer to the
+[extension generator](./Extension-generator.md) guide for creating an extension.
+
+## Usage
+
+In order to use the above component into our REST API application, we have a few
+more steps to go.
+
+- Add component to application.
+
+{% include code-caption.html content="/src/application.ts" %}
+
+```ts
+this.component(AuthenticationComponent);
+```
+
+- Add a step in custom sequence to check for authorization whenever any end
+  point is hit.
+
+{% include code-caption.html content="/src/sequence.ts" %}
+
+```ts
+import {inject} from '@loopback/context';
+import {
+  FindRoute,
+  InvokeMethod,
+  ParseParams,
+  Reject,
+  RequestContext,
+  RestBindings,
+  Send,
+  SequenceHandler,
+  HttpErrors,
+} from '@loopback/rest';
+import {
+  AuthorizatonBindings,
+  AuthorizeFn,
+  AuthorizeErrorKeys,
+} from './authorization';
+
+const SequenceActions = RestBindings.SequenceActions;
+
+export class MySequence implements SequenceHandler {
+  constructor(
+    @inject(SequenceActions.FIND_ROUTE) protected findRoute: FindRoute,
+    @inject(SequenceActions.PARSE_PARAMS) protected parseParams: ParseParams,
+    @inject(SequenceActions.INVOKE_METHOD) protected invoke: InvokeMethod,
+    @inject(SequenceActions.SEND) public send: Send,
+    @inject(SequenceActions.REJECT) public reject: Reject,
+    @inject(AuthorizatonBindings.AUTHORIZE_ACTION)
+    protected checkAuthorization: AuthorizeFn,
+  ) {}
+
+  async handle(context: RequestContext) {
+    try {
+      const {request, response} = context;
+      const route = this.findRoute(request);
+      const args = await this.parseParams(request, route);
+      // Do authentication of the user and fetch user permissions below
+      const permissions: string[] = [];
+      // This is main line added to sequence
+      // where we are invoking the authorize action function to check for access
+      const isAccessAllowed: boolean = await this.checkAuthorization(
+        permissions,
```

Comment:
Is `permissions` always an empty array? Does `checkAuthorization` add items to `permissions`?

Reply:
Again, I intentionally omitted the authentication logic as it is out of scope for this. But for clarity sake, I added a new section of 'Consideratons' mentioning some of the authentication related stuff as prior considerations for this document. 
Also added authenticate request function as part of actual implementation here for guidance. 
Hope that resolves your query.